paging/common/src/main/kotlin/androidx/paging/PageEvent.kt

```diff
@@ -188,25 +188,8 @@ internal sealed class PageEvent<T : Any> {
     }
 
     data class LoadStateUpdate<T : Any>(
-        val loadType: LoadType,
-        val fromMediator: Boolean,
-        val loadState: LoadState // TODO: consider using full state object here
+        val combinedLoadStates: CombinedLoadStates
     ) : PageEvent<T>() {
-        init {
-            // endOfPaginationReached for local refresh is driven by null values in next/prev keys.
-            require(
```

Comment:
Since we can no longer rely on this, we should create some tests around this case to ensure NotLoading can only be set within insert events except in the case of remote state?